ocaml/lambda/debuginfo.ml

```diff
@@ -143,21 +166,22 @@ type item = {
   dinfo_scopes: Scoped_location.scopes;
 }
 
-type t = { dbg : item list; }
+type t = { dbg : item list; assume_zero_alloc : bool }
 
 type alloc_dbginfo_item =
   { alloc_words : int;
     alloc_dbg : t }
 type alloc_dbginfo = alloc_dbginfo_item list
 
-let none = { dbg = []; }
+let none = { dbg = []; assume_zero_alloc = false }
 
-let is_none { dbg } =
+let is_none { dbg; assume_zero_alloc } =
+  if assume_zero_alloc then false else
```

Comment:
Can you elaborate on why this change is needed?

Reply:
I interpreted `is_none t` to return true if and only if  `t` is equal to `Debuginfo.none`, i.e., taking into account all fields of the `Debuginfo.t` record.  Did you expect `is_none`  to work only on the list of items? 
I updated the interface `debuginfo.mli` to make it clear when functions operate on the list of items (i.e.,g `dbg` field of `Debuginfo.t`) rather than the entire record. It is used only during Emit. I think it addresses your comment below as well about sharing `Debuginfo` in `Emitaux` below. Please let me know what you think about this version.